File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.js
/**
 * @param {string} s
 * @return {string}
 */
var removeDuplicates = function(s) {
    for (let i = 0; i < s.length; i++) {
        if (i === 0) continue;
        if (s[i] == s[i-1]) {
            if (i === 1) {
                s = s.slice(i+1);
                i = 0;
            } else {
                s = s.slice(0, i - 1) + s.slice(i+1);
                i -= 2;
            }
        }
    }
    return s;
};

File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.py
class Solution:
    def removeDuplicates(self, s: str) -> str:
        stack = []
        for i in range(len(s)):
            if len(stack) == 0 or s[i] != stack[-1]:
                stack.append(s[i])
            else:
                stack.pop()
        return "".join(stack)